src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -1567,6 +1570,72 @@ int wm_vuldet_send_cve_report(vu_report *report) {
     return retval;
 }
 
+int wm_vuldet_send_removed_cve_report (cJSON* j_vuln, scan_ctx_t* scan_ctx) {
+    int retval = OS_INVALID;
+    char header[OS_SIZE_256 + 1];
+    char alert_msg[OS_MAXSTR + 1];
+    int send_queue;
+    char* alert_title = NULL;
+    char* alert = NULL;
+    char* package_name = cJSON_GetStringValue(cJSON_GetObjectItem(j_vuln, "name"));
+    char* package_version = cJSON_GetStringValue(cJSON_GetObjectItem(j_vuln, "version"));
+    char* package_architecture = cJSON_GetStringValue(cJSON_GetObjectItem(j_vuln, "architecture"));
+    char* cve_id = cJSON_GetStringValue(cJSON_GetObjectItem(j_vuln, "cve"));
+    char* cve_reference = cJSON_GetStringValue(cJSON_GetObjectItem(j_vuln, "reference"));
+    char* cve_type = cJSON_GetStringValue(cJSON_GetObjectItem(j_vuln, "type"));
+    // Define time to sleep between messages sent
+    int usec = 1000000 / wm_max_eps;
+
+    os_calloc(OS_SIZE_512, sizeof(char), alert_title);
+    snprintf(alert_title, OS_SIZE_512, "%s affecting %s was eliminated", cve_id, package_name);
+
+    cJSON* j_alert = cJSON_CreateObject();
+    cJSON* j_alert_cve = cJSON_CreateObject();
+    cJSON *j_package = cJSON_CreateObject();
+    cJSON_AddItemToObject(j_alert, "vulnerability", j_alert_cve);
+    cJSON_AddItemToObject(j_alert_cve, "package", j_package);
+
+    if (j_alert && j_alert_cve && j_package) {
+        cJSON_AddStringToObject(j_package, "name", package_name);
+        cJSON_AddStringToObject(j_package, "version", package_version);
+        cJSON_AddStringToObject(j_package, "architecture", package_architecture);
+        cJSON_AddStringToObject(j_alert_cve, "cve", cve_id);
+        // To trigger the alert, the status is defined here
+        cJSON_AddStringToObject(j_alert_cve, "status", "Removed");
```

Comment:
Although this is the only one place where the `REMOVED` status is used, I would define it in the `wdb.h` header file together with the other statuses and use the definition here.

```c
#define VULN_CVES_STATUS_REMOVED "REMOVED"
```

Reply:
Probably this will require changing the string to capital letters in the rule.

Reply:
+1

Reply:
You are right, it is much better with a definition.
Fixed !